src/components/CustomFileManager.vue

```diff
@@ -310,113 +315,175 @@ export default class CustomFileManager extends Vue {
+    // Enqueue for batch resolution and return header-only chips for now
+    if (isDataset) {
+      this.batchQueueDatasetIds.add(selectable._id);
+    } else {
+      this.batchQueueConfigurationIds.add(selectable._id);
+    }
+    this.scheduleBatchResolve();
+
+    return { chips: ret, type };
+  }
+
+  scheduleBatchResolve() {
+    if (this.batchTimer !== null) {
+      window.clearTimeout(this.batchTimer);
+      this.batchTimer = null;
+    }
+    // Debounce to accumulate ids from the current render pass
+    this.batchTimer = window.setTimeout(() => this.flushBatchResolve(), 50);
+  }
+
+  async flushBatchResolve() {
+    const datasetIds = Array.from(this.batchQueueDatasetIds);
+    const configurationIds = Array.from(this.batchQueueConfigurationIds);
+    this.batchQueueDatasetIds.clear();
+    this.batchQueueConfigurationIds.clear();
+    this.batchTimer = null;
+
+    if (datasetIds.length === 0 && configurationIds.length === 0) {
+      return;
+    }
+
+    // Query mapping with names
+    let pairs: Array<{
+      datasetId: string;
+      configurationId: string;
+      datasetName?: string;
+      configurationName?: string;
+    }> = [];
+    try {
+      pairs = await this.store.api.mapDatasetViews({
+        datasetIds: datasetIds.length ? datasetIds : undefined,
+        configurationIds: configurationIds.length
+          ? configurationIds
+          : undefined,
+        includeNames: true,
+      } as any);
+    } catch (e) {
+      // If endpoint not available, fall back silently
+      return;
+    }
+
+    // Build additional chips per item
+    const addChipsForDataset = (
+      id: string,
+      chipText: string,
+      configurationId: string,
+    ) => {
+      const current = this.chipsPerItemId[id];
+      if (!current) return;
+      const extra: IChipAttrs = {
+        color: "#4baeff",
+        text: chipText,
+        to: this.clickableChips
+          ? { name: "configuration", params: { configurationId } }
+          : undefined,
       };
-      if (this.clickableChips) {
-        firstChip.to = {
-          name: "dataset",
-          params: { datasetId: selectable._id },
-        };
+      const chips = [...current.chips, extra];
+      Vue.set(this.chipsPerItemId, id, { chips, type: "dataset" });
+    };
+
+    const addChipsForConfig = (
+      id: string,
+      chipText: string,
+      datasetId: string,
+    ) => {
+      const current = this.chipsPerItemId[id];
+      if (!current) return;
+      const extra: IChipAttrs = {
+        color: "#e57373",
+        text: chipText,
+        to: this.clickableChips
+          ? { name: "dataset", params: { datasetId } }
+          : undefined,
+      };
+      const chips = [...current.chips, extra];
+      Vue.set(this.chipsPerItemId, id, { chips, type: "configuration" });
+    };
+
+    // Track missing names to optionally resolve via batchResources
+    const missingDatasetIds = new Set<string>();
+    const missingConfigIds = new Set<string>();
```

Comment:
I'm surprised, this should not be necessary ?  
Did you notice missing names ?

Reply:
There are definitely cases where someone removes the dataset but not the configuration.